C/common/string_utils.cpp

```diff
@@ -462,3 +462,71 @@ std::string StringAround(const std::string& str, unsigned int pos,
 	size_t	len = before + after;
 	return str.substr(start, len);
 }
+
+/**
+ * Replicate a character/substring within an string
+ *
+ * @param out StringToManage    string in which apply the search and replicate
+ * @param     StringToSearch    string to search and replicate
+ * @param     StringReplicate  substitution string
+ *
+ */
+void replicate(std::string& StringToManage,
```

Comment:
Replicate is a bad name for this. Probably replaceAll us better

Reply:
Here function is replicating a single character to double character. Also **StringReplaceAll** function is already there but will not work in this scenario. That's why I created new function. I will rename the function as **StringReplaceAllEx** to make it more intuitive.